app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,608 @@
+                {...itemProps}
+                theme={theme}
+                selectable={isMultiselect || false}
+                selected={itemSelected}
+            />
+        );
+    };
+
+    const renderUserItem = (itemProps: any): JSX.Element => {
+        const itemSelected = Boolean(multiselectSelected[itemProps.item.id]);
+
+        return (
+            <UserListRow
+                key={itemProps.id}
+                {...itemProps}
+                theme={theme}
+                selectable={isMultiselect || false}
+                user={itemProps.item}
+                teammateNameDisplay={teammateNameDisplay}
+                selected={itemSelected}
+            />
+        );
+    };
+
+    const renderSelectedOptions = (): React.ReactElement<any, string> | null => {
+        const selectedItems: any = Object.values(multiselectSelected);
+
+        let optionComponents: React.ReactElement<any, string> | null = null;
+
+        if (selectedItems.length > 0) {
+            optionComponents = (
+                <>
+                    <SelectedOptions
+                        theme={theme}
+                        selectedOptions={selectedItems}
+                        dataSource={dataSource}
+                        onRemove={handleRemoveOption}
+                    />
+                    <View style={style.separator}/>
+                </>
+            );
+        }
+
+        return optionComponents;
+    };
+
+    const listType = dataSource === ViewConstants.DATA_SOURCE_USERS ? SECTIONLIST : FLATLIST;
+    let rowComponent;
+    if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+        rowComponent = renderUserItem;
+    } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+        rowComponent = renderChannelItem;
+    } else {
+        rowComponent = renderOptionItem;
+    }
+
+    const selectedOptionsComponent = renderSelectedOptions();
+
+    return (
+        <SafeAreaView style={style.container}>
+            <View
+                testID='integration_selector.screen'
+                style={style.searchBar}
+            >
+                <SearchBar
+                    testID='selector.search_bar'
+                    placeholder={intl.formatMessage({id: 'search_bar.search', defaultMessage: 'Search'})}
+                    inputStyle={searchBarInput}
+                    placeholderTextColor={changeOpacity(theme.centerChannelColor, 0.5)}
+                    onChangeText={onSearch}
+                    autoCapitalize='none'
+                    keyboardAppearance={getKeyboardAppearanceFromTheme(theme)}
+                    value={term}
+                />
+            </View>
+
+            {selectedOptionsComponent}
+
+            <CustomList
+                data={customListData as DataType}
+                key='custom_list'
+                listType={listType}
+                loading={loading}
+                loadingComponent={renderLoading()}
+                noResults={renderNoResults}
```

Comment:
Some of the functions passed here (like `rowComponent` and `renderNoResults`) are regenerated on every render. Review those and add the needed useCallback.